inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/eltwise/eltwise_kernel_vload8.cpp

```diff
@@ -32,18 +32,27 @@ bool EltwiseKernel_vload8::Validate(const Params& params, const optional_params&
 
     const auto& ewParams = static_cast<const eltwise_params&>(params);
 
-    for (size_t i = 0; i < ewParams.inputs.size(); i++) {
-        if ((ewParams.inputs[i].GetLayout() == DataLayout::b_fs_yx_fsv16 && ewParams.inputs[i].Feature().v % 16 != 0) ||
-            (ewParams.inputs[i].GetLayout() == DataLayout::b_fs_zyx_fsv16 && ewParams.inputs[i].Feature().v % 16 != 0) ||
-            (ewParams.inputs[i].GetLayout() == DataLayout::b_fs_yx_fsv4 && ewParams.inputs[i].Feature().v % 8 != 0) ||
-            ewParams.inputs[i].GetLayout() == DataLayout::fs_b_yx_fsv32)
+        for (size_t i = 0; i < ewParams.inputs.size(); i++) {
+            const auto input_layout = ewParams.inputs[i].GetLayout();
+            const auto batch_size = ewParams.inputs[i].Batch().v;
+            const auto feature_size = ewParams.inputs[i].Feature().v;
+            if ((input_layout == DataLayout::b_fs_yx_fsv16 && feature_size % 16 != 0) ||
+                (input_layout == DataLayout::b_fs_zyx_fsv16 && feature_size % 16 != 0) ||
+                (input_layout == DataLayout::b_fs_yx_fsv4 && feature_size % 8 != 0) ||
+                input_layout == DataLayout::fs_b_yx_fsv32 ||
+                (input_layout == DataLayout::bs_fs_yx_bsv32_fsv16 && (feature_size % 16 != 0 || batch_size % 32 != 0)) ||
+                (input_layout == DataLayout::bs_fs_yx_bsv32_fsv32 && (feature_size % 32 != 0 || batch_size % 32 != 0)))
+                return false;
+        }
+        if ((ewParams.output.GetLayout() == DataLayout::b_fs_yx_fsv16 && ewParams.output.Feature().v % 16 != 0) ||
+            (ewParams.output.GetLayout() == DataLayout::b_fs_zyx_fsv16 && ewParams.output.Feature().v % 16 != 0) ||
+            (ewParams.output.GetLayout() == DataLayout::b_fs_yx_fsv4 && ewParams.output.Feature().v % 8 != 0) ||
+            ewParams.output.GetLayout() == DataLayout::fs_b_yx_fsv32 ||
+            (ewParams.output.GetLayout() == DataLayout::bs_fs_yx_bsv32_fsv16 &&
+                (ewParams.output.Feature().v % 16 != 0 || ewParams.output.Batch().v % 32 != 0)) ||
+            (ewParams.output.GetLayout() == DataLayout::bs_fs_yx_bsv32_fsv32 &&
+                (ewParams.output.Feature().v % 32 != 0 || ewParams.output.Batch().v % 32 != 0)))
```

Comment:
Isn't it necessary to check `b_fs_yx_fsv32`?

Reply:
It is not that I added. Do you want to remove that b_fs_yx_fsv32 checking code?

Reply:
Added

Reply:
As I shared in offline discussion, actually we can allow this kernel for not alinged layouts either by 1) fix computing logic of tensor size or 2) fix dispatch data for work items to run for meaningful positions 

So I'd like to just ask : is this change decided by a performance results showing that using simd8 kernel was not that much better than using ref kernel? Or just for the work priority?

Reply:
For tensors of shape 32x3x512x512, the performance between the two kernels is similar. ref kernel also has accuracy issues including bugs, this PR should be included. Can't you optimize the vload8 kernel after merging this PR?

Reply:
As mentioned offline, proper approach would be to optimize through fsv16 kernel, not vload8 kernel.